### DISCUSSION TITLE
Mcginley Dynamic

### DISCUSSION THREAD

**Discussion by myalgomate** [DaveSkender/Stock.Indicators#866] [Indicators]
Community discussion for [McGinley Dynamic](https://daveskender.github.io/Stock.Indicators/indicators/Dynamic/#content) moving average

**Comment by DaveSkender** [comment]
Looks like a good, reputable one.
Adding to backlog, see #867 for tracking.

Update: this was implemented in v2.2.0

**Reply by myalgomate** [reply]
Thanks a lot.
It helped me a lot

**Reply by myalgomate** [reply]
Hi Dave,
Hope you are doing well.

In regards to this Mcginley indicator, I was checking in more depth regarding the calculations.

There are 2 version of the same and both are official. I was using your formula and my value was not matching exactly with some of the charting platforms. i removed value of k and when calculated it matched 100% with another charting platform.

I would suggest you to implement the same as a configurable parameter so that it would help to other people in future.

McGinley Dynamic Formula and Calculation
The formula for the calculation of the McGinley Dynamic Indicator is as follows:

MDi = MDi-1 + [Close – MDi-1 / N × (Close / MDi-1)⁴ ]

Where:

MD – Current McGinley Dynamic
MDi-1 – Previous McGinley Dynamic
Close – Closing price
N – the MA period
At first glance, the formula might look complicated. However, the calculation of the indicator is pretty straightforward.

You may see another version of the formula with an added parameter “k” in the denominator in some textbooks. It looks like this:

MDi = MDi-1 + [Close – MDi-1 / k × N × (Close / MDi-1)⁴ ]

Where:

k – constant 60% of the selected period N
The thing that might confuse you a bit is the idea of “k” being a certain percentage of “N.”

The relationship between both parameters defines how closely the Dynamic indicator tracks the instrument’s price. For example, if you want to emulate a 20-day MA, you should set the “N” at half the period, or, in this case – 10. The “k” will then equal 60%.

The fact that “k” isn’t included in the universal formula is because its value is constant and adjusts automatically.

In the end, it all comes to the fact that you can customize the whole indicator through the value you set for “N.” No matter the formula.

The way the indicator is calculated defines one of its notable characteristics. It speeds up in bear markets and slows down in bull markets. The reason is that, in general, traders want to sell as quickly as possible when the market goes down and ride the wave as long as possible when it is going up.

**Reply by myalgomate** [reply]
Some reference

https://www.investopedia.com/terms/m/mcginley-dynamic.asp

https://blog.earn2trade.com/mcginley-dynamic-indicator/

at location of 6:50 min
https://www.youtube.com/watch?v=ALZcQYR7rRQ

**Reply by DaveSkender** [reply]
Sure, I can make it an optional parameter with a default value.  The [S&C magazine article](http://technical.traders.com/archive/articlefinal.asp?file=%5CV28%5CC03%5C045TWOM.pdf) goes into much more depth on explaining this one

> McGinley states that N should be 60% of the moving average you wish to emulate.

Unfortunately this is so old it’ll be nearly impossible to know if this `k` value was ever meant to be a variable from the original authors published works.  At any rate, no harm in making it a param, it doesn’t change the formula at all.
